### PR TITLE
Integrate cli_pkg, replace Travis by GitHub Actions and upload executables

### LINKED ISSUE #77: Use GitHub Actions to generate native executables for Windows, Linux and macOS
We could use GitHub Actions to generate native executables.
Reference: https://blog.dantup.com/2019/11/easily-compiling-dart-to-native-executables-for-windows-linux-macos-with-github-actions/

Instead of `DanTup/gh-actions/setup-dart@master`, we should use the "official" [`dart-lang/setup-dart@v1`](https://github.com/marketplace/actions/setup-dart-sdk).

When running `dart2native` on my local machine I get:

```
$ dart2native bin/linkcheck.dart -v -o build/linkcheck
Compiling /home/guillaume/workspace/opensource/linkcheck/bin/linkcheck.dart to /home/guillaume/workspace/opensource/linkcheck/build/linkcheck using format exe:
Generating AOT kernel dill.
Info: Compiling without sound null safety
Generating AOT snapshot.
Generating executable.
Marking binary executable.
```

```
$ ldd build/linkcheck                                 
	linux-vdso.so.1 (0x00007ffcc6b76000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f1924d6a000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f1924d47000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f1924bf8000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f1924a06000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f192541f000)
```

Is it possible to generate a statically linked binary?

### PR DESCRIPTION
**Pull request by Mogztter**
resolves #77

### CODE REVIEW

**Comment by Mogztter** [comment]
@filiph you can see the result here: https://github.com/Mogztter/linkcheck/runs/2713560742

**Comment by Mogztter** [comment]
And here's a release (performed on my fork): https://github.com/Mogztter/linkcheck/releases/tag/2.0.18 🎉

File: tool/grind.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import 'package:cli_pkg/cli_pkg.dart' as pkg;
+import 'package:grinder/grinder.dart';
+
+void main(List<String> args) {
+  pkg.name.value = "linkcheck";
+  pkg.humanName.value = "linkcheck";
+  // FIXME: for testing purposes, should be "filiph/linkcheck"
+  pkg.githubRepo.value = "mogztter/linkcheck";
```
Comment (Mogztter) [inline tool/grind.dart:8]:
```suggestion
  pkg.githubRepo.value = "filiph/linkcheck";
```

**Comment by Mogztter** [comment]
A few notes on release:

- it will create a release on GitHub based on the version defined in `pubspec.yaml` (not based on the tag version)
https://github.com/filiph/linkcheck/blob/6363429e6ea4ef2b1aea1d8e1da66c46bb1e5414/pubspec.yaml#L2

    In my case, the release process was triggered by pushing the tag `v2.0.21` but as you can see, `cli_pkg` created a new tag/release named `2.0.18`.

- it will automatically create release notes based on the https://github.com/filiph/linkcheck/blob/master/CHANGELOG.md file (which is super neat!)

    As you can see on the `2.0.18` release, `cli_pkg` automatically took the corresponding section from the changelog file: "Allow the latest pkg:args;".

- archives contain the binary and the license file

**Comment by filiph** [comment]
That's amazing, thank you!

I'm going to merge this. Can you confirm that the following is true? (I want to add it to the README file.)

```markdown
### Releasing a new version

1. Commit all your changes, including updates to `CHANGELOG`, 
   and including updating the version number in `pubspec.yaml`
   and `lib/linkcheck.dart`. Let's say your new version number is `3.4.13`.
   That number should be reflected in all three files.
2. Tag the last commit with the same version number.
   The tag starts with the letter `v` and continues with the semver number.
   In our case, it would be `v3.4.13`.
3. Push to `master`.

This will run the GitHub Actions script in `.github/workflows/release.yml`,
building binaries and placing a new release into
[github.com/filiph/linkcheck/releases](https://github.com/filiph/linkcheck/releases).
```

**Comment by Mogztter** [comment]
Looks good.

> Tag the last commit with the same version number.
   The tag starts with the letter `v` and continues with the semver number.
   In our case, it would be `v3.4.13`.

I believe that `cli_pkg` will create a new release/tag with the version defined in `pubspec.yaml`.
So we might need to remove the prefix `v` when releasing a new release (and update 
`on.push.tag`) or override the `pkg.version` in `the tool/grind.dart` to add a `v` prefix.

Otherwise, I think `cli_pkg` will create a release/tag named `3.4.13`.

**Comment by Mogztter** [comment]
@filiph Does it make sense? For reference, https://github.com/sass/dart-sass is not using the `v` prefix. 

However, if we do want to keep the`v` prefix then we need to:

1. override the `pkg.version` in the `tool/grind.dart` to add a `v` prefix.
2. use the `v` prefix in the `CHANGELOG` (otherwise `cli_pkg` might be confused)


And, if we do want to drop the `v` prefix, then we need to update `on.push.tag`:

https://github.com/filiph/linkcheck/blob/f42a301b4554fb8b02b775354580ba1122e892cb/.github/workflows/release.yml#L6

The release procedure becomes:

2. Tag the last commit with the same version number.
   In our case, it would be `3.4.13`.

Once this is done, would it be possible to publish a new version? I would love to get the latest changes  since v2.0.15+1: https://github.com/filiph/linkcheck/compare/v2.0.15%2B1...master

Thanks!

Let me know if there's anything I can do to make that happen 😃

**Comment by filiph** [comment]
FWIW, I like the `v` prefix, and it would be consistent with the git tag history, but I don't have a strong preference.

Would you be able to make either change (`v` or no-`v`)?

**Comment by Mogztter** [comment]
> FWIW, I like the v prefix, and it would be consistent with the git tag history, but I don't have a strong preference.

Me too but I took a closer look at `cli_pkg` and it seems that it's not currently supported to use a version with a `v` prefix.

> Would you be able to make either change (v or no-v)?

Sure, I will implement the no-v solution 😉

**Comment by filiph** [comment]
Thank you!

**Comment by Mogztter** [comment]
No worries, the follow-up pull request is here: https://github.com/filiph/linkcheck/pull/80

**Pull request merged**